SPHINXsys/src/for_3D_build/geometries/geometry.cpp

```diff
@@ -124,22 +129,23 @@ namespace SPH
 		for (int i = 0; i != number_of_vertices; ++i)
 		{
 			Vec3d vertex_position = triangle_mesh_->getVertexPosition(i);
-			for (int j = 0; j != 3; ++j) {
+			for (int j = 0; j != 3; ++j)
+			{
 				lower_bound[j] = SMIN(lower_bound[j], vertex_position[j]);
 				upper_bound[j] = SMAX(upper_bound[j], vertex_position[j]);
 			}
 		}
 		return BoundingBox(lower_bound, upper_bound);
 	}
 	//=================================================================================================//
-	bool ComplexShape::checkContain(Vec3d& input_pnt, bool BOUNDARY_INCLUDED)
+	bool ComplexShape::checkContain(const Vec3d &input_pnt, bool BOUNDARY_INCLUDED)
 	{
 		bool exist = false;
 		bool inside = false;
-		
-		for (auto& each_shape : triangle_mesh_shapes_)
+
+		for (auto &each_shape : triangle_mesh_shapes_)
 		{
-			TriangleMeshShape* sp = each_shape.first;
+			TriangleMeshShape *sp = each_shape.first;
```

Comment:
Can we start refactoring the pointers to shared and unique pointers next time we have some changes for these pointers?

Reply:
This is a good idea as smart pointers handle the creation and destruction of the object (replaces malloc/free new/delete).
We have them throughout the GPU version. 

The only thing that I am concerned about is weather the compiling units involved with SimBody are compatible with these advanced C++ features or not? In particular, the std::unique_ptr which has been introduced in C++14. The std::shared_ptr should be fine, as they are already shipped with C++11.

Reply:
Yes. My suggestion is use smart pointer for those non-performance related pointers. However, I would not suggest to use it for computing intensive cases.